### PR TITLE
TYP: make dtype required in _from_sequence_of_strings

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
> Error: /home/runner/work/pandas/pandas/pandas/core/arrays/base.py:334:EX03:pandas.api.extensions.ExtensionArray._from_sequence_of_strings:flake8 error: line 2, col 4: E121 continuation line under-indented for hanging indent

@mroeschke is this check new?  i dont see what it is complaining about

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -354,7 +354,8 @@ def _from_sequence_of_strings(
 
         Examples
         --------
-        >>> pd.arrays.IntegerArray._from_sequence_of_strings(["1", "2", "3"])
+        >>> pd.arrays.IntegerArray._from_sequence_of_strings(
+        ...    ["1", "2", "3"], dtype=pd.Int64Dtype())
```
Comment (mroeschke) [inline pandas/core/arrays/base.py:358]:
```suggestion
        >>> pd.arrays.IntegerArray._from_sequence_of_strings(
        ...     ["1", "2", "3"], dtype=pd.Int64Dtype()
        ... )
```

I _think_ this should fix it?

**Comment by jbrockmendel** [comment]
good suggestion, seems to have done the trick

File: pandas/core/arrays/arrow/array.py [hunk 2]
```diff
@@ -316,7 +318,7 @@ def _from_sequence(
 
     @classmethod
     def _from_sequence_of_strings(
-        cls, strings, *, dtype: Dtype | None = None, copy: bool = False
+        cls, strings, *, dtype: ExtensionDtype, copy: bool = False
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:321]: At least for arrow, `dtype=None` was supposed to mean "let `pa.array` infer the type, and now that can't be done?

**Comment by jbrockmendel** [comment]
This method is only ever called with a style object. There’s also a recent
issue about having a dedicated constructor for inference.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**